python/ray/_private/usage/usage_lib.py

```diff
@@ -237,6 +238,37 @@ def _put_library_usage(library_usage: str):
             logger.debug(f"Failed to write a library usage to the home folder, {e}")
 
 
+def record_extra_usage_tag(key: str, value: str):
```

Comment:
Could we actually make "key" an enum in a whitelist ALLOWED_KEYS? The main thing is we don't want keys to be too high cardinality, otherwise this could break the data pipeline schema. E.g.,

```
class TagKey(Enum):
   # tf / torch / jax
   RLLIB_FRAMEWORK = 1
   ...
```

Reply:
Discussed offline: extra tags will be stored as map in delta lake so high cardinality shouldn't be a problem as long as it's not too crazy.

Reply:
After a second thought,  I think it's still a good idea to have an allow list:
1. Make sure the uniqueness
2. Allow us to audit the usage as code owner to make sure nothing too crazy happens
3. We can always relax it later if needed.